core/dispatcher/src/main/scala/whisk/core/container/WhiskContainer.scala

```diff
@@ -127,27 +139,59 @@ class WhiskContainer(
      * @param msg the message to post
      * @return response from container if any as array of byte
      */
-    private def sendPayload(endpoint: String, msg: JsObject, timeout: Int): RunResult = {
+    private def sendPayload(endpoint: String, msg: JsObject, timeout: FiniteDuration)(implicit system: ActorSystem): RunResult = {
+        import system.dispatcher
+
         val start = ContainerCounter.now()
-        val result = containerIP map { host =>
-            try {
-                val connection = HttpUtils.makeHttpClient(timeout, true)
-                val http = new HttpUtils(connection, host)
-                val (code, bytes) = http.dopost(endpoint, msg, Map(), timeout)
-                Try { connection.close() }
-                val returnCode = if (code == -1) ActivationResponse.ContainerError else code
```

Comment:
change might not be harmful but I think makes this statement in the invoker out of date:
`                    val removeContainer = failedInit || response.exists { _._1 == ActivationResponse.ContainerError }`

since the second term cannot be true anymore.
confirm? @markusthoemmes @psuter